app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -161,8 +161,13 @@ class PendingTransactionTracker extends EventEmitter {
 
       return
     }
-    // *note to self* hard failure point
-    const transactionReceipt = await this.query.getTransactionReceipt(txHash)
+
+    let transactionReceipt
+    try {
+      transactionReceipt = await this.query.getTransactionReceipt(txHash)
+    } catch (_) {
+      transactionReceipt = undefined
```

Comment:
A value of `undefined` seems like a strange way to signal that a failure occurred. You could use a separate variable to indicate that a failure had occurred, to disambiguate the error case from when `getTransactionReceipt` happens to return `undefined`.